client/src/pages/Settings/Availability/DateForm/DateForm.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import Box from '@material-ui/core/Box';
+import useStyles from './useStyles';
+import Typography from '@material-ui/core/Typography';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+
+interface date {
+  to: string;
+  from: string;
+}
+interface Availability {
+  [key: string]: date;
+}
+
+interface Props {
+  date: string;
+  availability: Availability;
+  handleTimeFrom: (event: React.ChangeEvent<{ value: any }>, date: string) => void;
+  handleTimeTo: (event: React.ChangeEvent<{ value: any }>, date: string) => void;
+}
+
+const times = [
+  '',
+  '00:00',
+  '01:00',
+  '02:00',
+  '03:00',
+  '04:00',
+  '05:00',
+  '06:00',
+  '07:00',
+  '08:00',
+  '09:00',
+  '10:00',
+  '11:00',
+  '12:00',
+  '13:00',
+  '14:00',
+  '15:00',
+  '16:00',
+  '17:00',
+  '18:00',
+  '19:00',
+  '20:00',
+  '21:00',
+  '22:00',
+  '23:00',
+];
```

Comment:
Feel like this can be generated computationally

Reply:
OK